examples/conversation_with_agent_with_finetuned_model/huggingface_model.py

```diff
@@ -0,0 +1,589 @@
+                    f"'{pretrained_model_name_or_path}'"
+                    f" from '{local_model_path}'",
+                )
+            self.tokenizer.add_special_tokens({"pad_token": "[PAD]"})
+
+        except Exception as e:
+            # Handle exceptions during model loading,
+            # such as file not found or load errors
+            error_message = (
+                f"Failed to load tokenizer for model"
+                f" '{pretrained_model_name_or_path}' from "
+                f"'{local_model_path}': {e}"
+            )
+            logger.error(error_message)
+
+            raise
+
+    def fine_tune(
+        self,
+        data_path: Optional[str] = None,
+        output_dir: Optional[str] = None,
+        fine_tune_config: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """
+        Fine-tune the agent's model using data from the specified path.
+
+        Arguments:
+            data_path (str): The file path to the training
+                             data from Hugging Face.
+            output_dir (str, optional): User specified path
+                                       to save the fine-tuned model
+                                       and its tokenizer. By default
+                                       save to this example's
+                                       directory if not specified.
+
+        Raises:
+            Exception: If the fine-tuning process fails. This could be
+            due to issues with the data path, configuration parameters,
+            or internal errors during the training process.
+        """
+        try:
+            self.model = self.fine_tune_training(
+                self.model,
+                self.tokenizer,
+                data_path,
+                output_dir,
+                token=self.huggingface_token,
+                fine_tune_config=fine_tune_config,
+            )
+
+            logger.info(
+                f"Successfully fine-tuned model with data from '{data_path}'",
+            )
+        except Exception as e:
+            logger.error(
+                f"Failed to fine-tune model with data from '{data_path}': {e}",
+            )
+            raise
+
+    def formatting_prompts_func(
+        self,
+        example: Dict[str, List[List[str]]],
+    ) -> List[str]:
+        """
+        Formats each conversation in the dataset for training.
+        Args:
+            example (Dict[str, List[List[str]]]): A dataset.
+
+        Returns:
+            List[str]: A dataset with combined field.
+        """
+        output_texts = []
+        for i in range(len(example["conversations"])):
+            text = (
+                "### Question: "
+                + example["conversations"][i][0]
+                + "\n ### Answer: "
+                + example["conversations"][i][1]
+            )
+            output_texts.append(text)
+        return output_texts
+
+    def fine_tune_training(
```

Comment:
Prefix the function that not exposed to users with "_"

Reply:
Done. But I didn't prefix `format` with an underscore as this is the case for all wrappers under `agentscope/src/agentscope/models`. Is `format` intended to be exposed to the users?